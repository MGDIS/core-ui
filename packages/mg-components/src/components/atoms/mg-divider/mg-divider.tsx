import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'mg-divider',
  styleUrl: '../../../../node_modules/@mgdis/styles/dist/components/mg-divider.css',
  shadow: true,
})
export class MgDivider {
  /**************
   * Decorators *
   **************/

  /**
   * Define component size
   */
  @Prop() size: 'regular' | 'full' = 'regular';

  /**
   * Render
   * @returns HTML Element
   */
  render(): HTMLElement {
    return (
      <div
        class={{
          'mg-divider': true,
          'mg-divider--full': this.size === 'full',
        }}
      ></div>
    );
  }
}
