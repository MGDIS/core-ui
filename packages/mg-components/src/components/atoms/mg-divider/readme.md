# mg-divider



<!-- Auto Generated Below -->


## Properties

| Property    | Attribute    | Description           | Type      | Default |
| ----------- | ------------ | --------------------- | --------- | ------- |
| `fullWidth` | `full-width` | Define component size | `boolean` | `false` |


## CSS Custom Properties

| Name                              | Description                                                                                                                      |
| --------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- |
| `--mg-c-divider-color-background` | Defines the background color of the divider. The default value is `color-mix(in srgb, var(--mg-b-color-dark), transparent 85%)`. |
| `--mg-c-divider-spacing-vertical` | Defines the vertical spacing of the divider. The default value is `--mg-b-size-40`.                                              |
| `--mg-c-divider-thickness`        | Defines the thickness of the divider. The default value is `--mg-b-size-border`.                                                 |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
