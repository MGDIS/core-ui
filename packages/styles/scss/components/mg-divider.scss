/*!
 * @prop --mg-c-divider-background-color: Defines the background color of the divider. The default value is `color-mix(in srgb, var(--mg-b-color-dark), transparent 85%)`.
 * @prop --mg-c-divider-thickness: Defines the thickness of the divider. The default value is `--mg-b-size-border`.
 * @prop --mg-c-divider-vertical-spacing: Defines the vertical spacing of the divider. The default value is `--mg-b-size-40`.
 */

.mg-c-divider {
  width: var(--mg-b-size-120);
  height: var(--mg-c-divider-thickness);
  margin: var(--mg-c-divider-vertical-spacing) auto;
  background-color: var(--mg-c-divider-background-color);

  &.mg-c-divider--full-width {
    width: 100%;
  }
}
