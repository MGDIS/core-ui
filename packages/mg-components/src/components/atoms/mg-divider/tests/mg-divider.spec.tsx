import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgDivider } from '../mg-divider';

const getPage = args =>
  newSpecPage({
    components: [MgDivider],
    template: () => <mg-divider {...args}></mg-divider>,
  });

describe('mg-divider', () => {
  test.each([false, true])('full-width %s', async fullWidth => {
    const { root } = await getPage({ fullWidth });
    expect(root).toMatchSnapshot();
  });
});
