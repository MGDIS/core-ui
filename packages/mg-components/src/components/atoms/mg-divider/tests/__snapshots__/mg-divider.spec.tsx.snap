// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-divider full-width false 1`] = `
<mg-divider>
  <mock:shadow-root>
    <div class="mg-c-divider"></div>
  </mock:shadow-root>
</mg-divider>
`;

exports[`mg-divider full-width true 1`] = `
<mg-divider>
  <mock:shadow-root>
    <div class="mg-c-divider mg-c-divider--full-width"></div>
  </mock:shadow-root>
</mg-divider>
`;
